iowrappers/redis_client.go

```diff
@@ -151,26 +151,35 @@ func (r *RedisClient) StorePlacesForLocation(context context.Context, geocodeInS
 	return nil
 }
 
-func (r *RedisClient) SetPlacesOnCategory(context context.Context, places []POI.Place) {
+// SetPlacesAddGeoLocations() is actively being used
+// It stores two types of information in redis
+// 1. key-value pair, {placeID: POI.place}
+// 2. add place to the correct bucket in geohashing for nearbysearch()
+func (r *RedisClient) SetPlacesAddGeoLocations(c context.Context, places []POI.Place) {
 	wg := &sync.WaitGroup{}
-	wg.Add(len(places))
+	wg.Add(2 * len(places)) // start two threads for each place
 	for _, place := range places {
-		placeCategory := POI.GetPlaceCategory(place.LocationType)
-		geolocation := &redis.GeoLocation{
-			Name:      place.ID,
-			Latitude:  place.GetLocation().Latitude,
-			Longitude: place.GetLocation().Longitude,
-		}
-		redisKey := "placeIDs:" + strings.ToLower(string(placeCategory))
-		_, cmdErr := r.client.GeoAdd(context, redisKey, geolocation).Result()
-
-		utils.LogErrorWithLevel(cmdErr, utils.LogError)
-
-		r.setPlace(context, place, wg)
+		go r.GeoAddWorkerByCategory(c, place, wg)
+		go r.setPlace(c, place, wg)
 	}
 	wg.Wait()
 }
 
+// GeoAddWorkerByCategory adds geo-locaiton of the place,
+// keys are identified with different place categories
+func (r *RedisClient) GeoAddWorkerByCategory(c context.Context, place POI.Place, wg *sync.WaitGroup) {
+	defer wg.Done()
+	placeCategory := POI.GetPlaceCategory(place.LocationType)
+	geoLocation := &redis.GeoLocation{
+		Name:      place.ID,
+		Latitude:  place.GetLocation().Latitude,
+		Longitude: place.GetLocation().Longitude,
+	}
+	redisKey := "placeIDs:" + strings.ToLower(string(placeCategory))
```

Comment:
@timwangmusic  is this the part you were mentioning about increasing the number of buckets we store geolocations? 

Currently we just split by visit and eatory IIUC. this leads to an issue where the retrieved results for restaurants can have limited number of "$$$" ones.

Reply:
This is not the part I mentioned. The matching package requests the number of results for a category, see `NearbySearchForCategory` method.

Reply:
sg, thanks!